Converter/app/src/main/java/com/javalabs/converter/DataFragment.java

```diff
@@ -0,0 +1,202 @@
+                        Toast.makeText(getActivity(), "Position = " + position, Toast.LENGTH_SHORT).show();
+                        break;
+                }
+            }
+            @Override
+            public void onNothingSelected(AdapterView<?> arg0) {
+            }
+        });
+
+    }
+
+    private void setSpinnerToListener(){
+        spinnerTo.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @SuppressLint("NonConstantResourceId")
+            @Override
+            public void onItemSelected(AdapterView<?> parent, View view,
+                                       int position, long id) {
+                mViewModel.selectMeasurementTo(parent.getItemAtPosition(position).toString(), position);
+                mViewModel.update();
+            }
+            @Override
+            public void onNothingSelected(AdapterView<?> arg0) {
+            }
+        });
+    }
+
+    private void setSpinnerFromListener(){
+        spinnerFrom.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @SuppressLint("NonConstantResourceId")
+            @Override
+            public void onItemSelected(AdapterView<?> parent, View view,
+                                       int position, long id) {
+                mViewModel.selectMeasurementFrom(parent.getItemAtPosition(position).toString(), position);
+                mViewModel.update();
+            }
+            @Override
+            public void onNothingSelected(AdapterView<?> arg0) {
+            }
+        });
+    }
+
+    private void changeSpinner(int category_unit){
+        ArrayAdapter<?> adapter_date_from = ArrayAdapter.createFromResource(getContext(),
+                                                                                category_unit,
+                                                                                R.layout.spinner_item);
+        adapter_date_from.setDropDownViewResource(R.layout.dropdown_item);
+        spinnerFrom.setAdapter(adapter_date_from);
+        spinnerTo.setAdapter(adapter_date_from);
+    }
+
+    public void copyInput() {
+        ClipboardManager clipboard = (ClipboardManager) getActivity().getSystemService(Context.CLIPBOARD_SERVICE);
+            ClipData clip = ClipData.newPlainText(getString(R.string.input_value),
+                                                    mViewModel.getSelectedDataFrom().getValue());
+        clipboard.setPrimaryClip(clip);
+        Toast.makeText(getActivity(), R.string.toast_copy, Toast.LENGTH_SHORT).show();
+    }
+
+    public void copyOutput() {
+        ClipboardManager clipboard = (ClipboardManager) getActivity().getSystemService(Context.CLIPBOARD_SERVICE);
+        ClipData clip = ClipData.newPlainText(getString(R.string.output_value), mViewModel.getSelectedDataTo().getValue());
+        clipboard.setPrimaryClip(clip);
+        Toast.makeText(getActivity(), R.string.toast_copy, Toast.LENGTH_SHORT).show();
+    }
+}
```

Comment:
According to the requirements, these functions should be present only in one product flavour, so it is better to remove them from the `main` source set (which is used in all flavours).

Then you can create another fragment (just for premium flavour) which will extend this one and enrich it with additional functionality.

Can you do this?)

Reply:
I divided DataFragment into different flavors.  I hope this is suitable for your request)

Reply:
Almost. Now you have a lot of identical code in both fragments.

> Then you can create another fragment (just for premium flavour) which will extend this one and enrich it with additional functionality.

Is it clear what I mean here?

Reply:
I made a new fragment, please see if it is correct